cluster.tf

```diff
@@ -113,7 +113,7 @@ resource "aws_security_group_rule" "cluster_private_access_sg_source" {
 
 resource "aws_iam_role" "cluster" {
   count                 = var.manage_cluster_iam_resources && var.create_eks ? 1 : 0
-  name_prefix           = var.cluster_iam_role_name != "" ? null : var.cluster_name
+  name_prefix           = var.cluster_iam_role_name == "" ? var.cluster_name : null
```

Comment:
To me, both writing are similar.

1. `name_prefix = var.cluster_iam_role_name != "" ? null : var.cluster_name` 
If  `var.cluster_iam_role_name`  is not empty, then return `null` otherwise return the cluster name
=> This mean also: return null if `var.cluster_iam_role_name` is empty (similar to 2)

2. `name_prefix  = var.cluster_iam_role_name == "" ? var.cluster_name : null`
If `var.cluster_iam_role_name` is empty , then return cluster name or `null` otherwise.
=> This mean also, return null if `var.cluster_iam_role_name` is not empty (similar to 1).

Please see:

```shell
> "x" != "" ? null : "y"
tostring(null)
> "" != "" ? null : "y"
"y"
> "x" == "" ? "y" : null
tostring(null)
> "" == "" ? "y" : null
"y"
```

Reply:
they are same, but if you will take a look on `name` conditional this is issue of problem.

Reply:
```hcl
  name_prefix           = var.cluster_iam_role_name != "" ? null : var.cluster_name
  name                  = var.cluster_iam_role_name != "" ? var.cluster_iam_role_name : null
```

If we managed IAM resources (`var.manage_cluster_iam_resources=true`) and `var.cluster_iam_role_name` is defined, then use `var.cluster_iam_role_name` as role name and don't use `name_prefix`. Otherwise set the `name_prefix`.

This is what I read. I'm probably missing something.

Reply:
Actually, the `var.worker_role_name` behave the same. See https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/workers.tf#L448-L449

Reply:
ahhh, yup you are right, closing PR